sdk/storage/Azure.Storage.Common/src/Shared/StorageResponseClassifier.cs

```diff
@@ -23,10 +22,24 @@ public StorageResponseClassifier(Uri secondaryStorageUri)
         /// <returns></returns>
         public override bool IsRetriableResponse(HttpMessage message)
         {
-            if (message.Request.Uri.Host == SecondaryStorageUri.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
+            if (message.Request.Uri.Host == SecondaryStorageUri?.Host && message.Response.Status == Constants.HttpStatusCode.NotFound)
             {
                 return true;
             }
+
+            // Retry select Storage Error Codes
+            if (message.Response.Status >= 400 &&
+                message.Response.Headers.TryGetValue("x-ms-error-code", out var error))
+            {
+                switch (error)
+                {
+                    case "InternalError":
+                    case "OperationTimedOut":
+                    case "ServerBusy":
+                    case "CannotVerifyCopySource":
```

Comment:
The top 3 are fine, but I don't know about `CannotVerifyCopySource`.  That's one I didn't care about for testing, but I'm not as comfortable with in product code.

Reply:
Looks like you beat me to it